### ISSUE TITLE
Frozen TTY's due to machinectl

### ISSUE THREAD

**Issue by shibumi** [systemd/systemd#2061]
If i try `machinectl shell root@.host` on my Archlinux Host `machinectl` crashes `Xorg Server`.

systemd version:

```
systemd 228
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +ID
```

`uname -a`:

```
Linux trudy 4.2.5-1-ARCH #1 SMP PREEMPT Tue Oct 27 08:28:41 CET 2015 i686 GNU/Linux
```

**Comment by poettering** [comment]
This is not a useful bug report. What does "crash" mean in this case? Why do you report this against systemd, and not Xorg? Do you have a backtrace?

Since it's xorg, not systemd that is crashing here, will close this now.

**Comment by shibumi** [comment]
Sorry maybe I should be more precisely. i was in hurry. 

I played a bit around with this bug @poettering and it seems that Xorg is not crashing.

I do the following:

I am in TTY1 .. I start DWM.. I start urxvt as favourite terminal emulator. Now I type in: `machinectl shell root@.host` .. my screen is black now. 

Now I am switching to TTY2 via CTRL+ALT+F2. I switch back to TTY1 via CTRL+ALT+F1 and voila.. everything is normal and I am logged in as root via machinectl. 

Now I try to switch back to TTY2 but I see just a frozen screen of my DWM (window manager).. I checked every other TTY. On every other TTY I just see a frozen DWM session.

During this process Xorg is never crashing. The process is still running. I just thought it's crashing because I got the black screen after `machinectl shell root.host`. 

I don't think that this is a xorg bug. Why has this bug effects on all other TTY's?
Therefore I have no stacktrace sorry.

**Comment by poettering** [comment]
if VT switching is borked, that's between X and the graphics drivers, systemd is not really involed: it tracks VT state, but generally doesn't alter it (unless explicitly asked).

Is there anything in the logs about anything being weird?

**Comment by shibumi** [comment]
nothing special. There was just a coredump of chrome because of the 'blackscreen' I think. But maybe you can find more:

```
Nov 30 19:00:43 trudy systemd[26485]: pam_unix(systemd-user:session): session opened for user
Nov 30 19:00:43 trudy systemd[1]: Started Session c3 of user root.
Nov 30 19:00:43 trudy systemd[26485]: Listening on Sound System.
Nov 30 19:00:43 trudy systemd[26485]: Reached target Timers.
Nov 30 19:00:43 trudy systemd[26485]: Starting D-Bus User Message Bus Socket.
Nov 30 19:00:43 trudy systemd[26485]: Reached target Paths.
Nov 30 19:00:43 trudy systemd[26485]: Listening on D-Bus User Message Bus Socket.
Nov 30 19:00:43 trudy systemd[26485]: Reached target Sockets.
Nov 30 19:00:43 trudy systemd[26485]: Reached target Basic System.
Nov 30 19:00:43 trudy systemd[26485]: Reached target Default.
Nov 30 19:00:43 trudy systemd[26485]: Startup finished in 143ms.
Nov 30 19:00:43 trudy systemd[1]: Started User Manager for UID 0.
Nov 30 19:00:55 trudy polkitd[1567]: Unregistered Authentication Agent for unix-process:26474
Nov 30 19:00:55 trudy org.freedesktop.Notifications[629]: XIO:  fatal IO error 11 (Resource t
Nov 30 19:00:55 trudy org.freedesktop.Notifications[629]: after 15608 requests (15608 known p
Nov 30 19:00:55 trudy systemd-logind[438]: Removed session c3.
Nov 30 19:00:55 trudy systemd[26485]: Stopped target Default.
Nov 30 19:00:55 trudy systemd[26485]: Stopped target Basic System.
Nov 30 19:00:55 trudy systemd[26485]: Stopped target Timers.
Nov 30 19:00:55 trudy systemd[26485]: Stopped target Paths.
Nov 30 19:00:55 trudy systemd[26485]: Stopped target Sockets.
Nov 30 19:00:55 trudy systemd[26485]: Closed Sound System.
Nov 30 19:00:55 trudy systemd[26485]: Closed D-Bus User Message Bus Socket.
Nov 30 19:00:55 trudy systemd[26485]: Reached target Shutdown.
Nov 30 19:00:55 trudy systemd[1]: Stopping User Manager for UID 0...
Nov 30 19:00:55 trudy systemd[26485]: Starting Exit the Session...
Nov 30 19:00:55 trudy org.a11y.atspi.Registry[16430]: XIO:  fatal IO error 11 (Resource tempo
Nov 30 19:00:55 trudy org.a11y.atspi.Registry[16430]: after 1019 requests (1019 known process
Nov 30 19:00:55 trudy systemd[26485]: Received SIGRTMIN+24 from PID 26505 (kill).
Nov 30 19:00:55 trudy systemd[26487]: pam_unix(systemd-user:session): session closed for user
Nov 30 19:00:55 trudy systemd[1]: Stopped User Manager for UID 0.
Nov 30 19:00:55 trudy systemd[1]: Removed slice user-0.slice.
Nov 30 19:01:01 trudy systemd-coredump[26511]: Process 26510 (chrome) of user 1000 dumped cor

                                               Stack trace of thread 26510:
                                               #0  0x00000000b7716bc8 __kernel_vsyscall (linu
                                               #1  0x00000000b613ee56 raise (libc.so.6)
                                               #2  0x00000000b61404d7 abort (libc.so.6)
                                               #3  0x0000000080a811d9 n/a (chrome)
Nov 30 19:01:01 trudy CROND[26529]: (root) CMD (run-parts /etc/cron.hourly)
Nov 30 19:01:02 trudy anacron[26535]: Anacron started on 2015-11-30
Nov 30 19:01:02 trudy anacron[26535]: Will run job `cron.daily' in 9 min.
Nov 30 19:01:02 trudy anacron[26535]: Jobs will be executed sequentially
Nov 30 19:01:08 trudy upowerd[5697]: (upowerd:5697): GLib-GObject-WARNING **: value "-0.67749
Nov 30 19:02:40 trudy polkitd[1567]: Registered Authentication Agent for unix-process:26921:4
Nov 30 19:02:42 trudy polkitd[1567]: Operator of unix-process:26921:44872828 successfully aut
Nov 30 19:02:42 trudy systemd[1]: Started Shell for User root.
Nov 30 19:02:42 trudy systemd[26930]: pam_unix(login:session): session opened for user root b
Nov 30 19:02:42 trudy systemd[1]: Created slice user-0.slice.
Nov 30 19:02:42 trudy systemd[1]: Starting User Manager for UID 0...
Nov 30 19:02:42 trudy systemd-logind[438]: New session c4 of user root.
Nov 30 19:02:42 trudy systemd[1]: Started Session c4 of user root.
Nov 30 19:02:42 trudy systemd[26931]: pam_unix(systemd-user:session): session opened for user
Nov 30 19:02:42 trudy systemd[26931]: Reached target Timers.
Nov 30 19:02:42 trudy systemd[26931]: Reached target Paths.
Nov 30 19:02:42 trudy systemd[26931]: Listening on Sound System.
Nov 30 19:02:42 trudy systemd[26931]: Starting D-Bus User Message Bus Socket.
Nov 30 19:02:43 trudy systemd[26931]: Listening on D-Bus User Message Bus Socket.
Nov 30 19:02:43 trudy systemd[26931]: Reached target Sockets.
Nov 30 19:02:43 trudy systemd[26931]: Reached target Basic System.
Nov 30 19:02:43 trudy systemd[26931]: Reached target Default.
Nov 30 19:02:43 trudy systemd[26931]: Startup finished in 56ms.
Nov 30 19:02:43 trudy systemd[1]: Started User Manager for UID 0.
Nov 30 19:02:45 trudy polkitd[1567]: Unregistered Authentication Agent for unix-process:26921
Nov 30 19:02:45 trudy systemd-logind[438]: Removed session c4.
Nov 30 19:02:45 trudy systemd[26931]: Stopped target Default.
Nov 30 19:02:45 trudy systemd[26931]: Stopped target Basic System.
Nov 30 19:02:45 trudy systemd[26931]: Stopped target Timers.
Nov 30 19:02:45 trudy systemd[26931]: Stopped target Sockets.
Nov 30 19:02:45 trudy systemd[26931]: Closed Sound System.
Nov 30 19:02:45 trudy systemd[26931]: Closed D-Bus User Message Bus Socket.
Nov 30 19:02:45 trudy systemd[26931]: Reached target Shutdown.
Nov 30 19:02:45 trudy systemd[26931]: Starting Exit the Session...
Nov 30 19:02:45 trudy systemd[26931]: Stopped target Paths.
Nov 30 19:02:45 trudy systemd[26931]: Received SIGRTMIN+24 from PID 26954 (kill).
Nov 30 19:02:45 trudy systemd[26933]: pam_unix(systemd-user:session): session closed for user
Nov 30 19:02:45 trudy systemd[1]: Stopping User Manager for UID 0...
Nov 30 19:02:45 trudy systemd[1]: Stopped User Manager for UID 0.
Nov 30 19:02:45 trudy systemd[1]: Removed slice user-0.slice.
```

**Comment by webczat** [comment]
see issue #2198
Seems to be systemd bug.

**Comment by shibumi** [comment]
@webczat #2198 links to #2061 (this one here).. I am really unsure if this is a systemd bug. Maybe we should wait for @poettering

**Comment by webczat** [comment]
yes, it links here, that is why I added a back reference because I believe it is the same thing. My testing actually showed it cannot be a xorg bug, this thing happens precisely after machinectl shell and it is actually caused by the tty being switched to the utf8 keyboard mode. Maybe something resets it.

**Comment by webczat** [comment]
I have now discovered the cause of the bug. It is very simple:
In src/machine/machine-dbus.c, in function bus_machine_method_open_shell(), while creating a transient unit, the function adds a TTYReset property and sets it to true.
I believe that causes a tty reset before startup and after stop of the unit, and because TTYPath is not set, it defaults to /dev/console and actually resets the tty where x is running. A slave pseudoterminal is destroyed after the unit stops I believe, and created before unit starts, so there is probably no point in resetting it.

**Comment by poettering** [comment]
Proper fix waiting in #2463
